### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Richa-rani02**
Added login/logout functionality with Toast

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -1,36 +1,31 @@
...
+    <header>
+      <div className="left_area">
         <label for="check">
-      <GiHamburgerMenu className="hamburger-icon"/>
-      </label>
-      <div>
-      {/* <img className="logo"
-          src="https://res.cloudinary.com/dgomw715r/image/upload/v1648184197/ProjectImages/noteslogo_qenwcd.png"
-          alt="logo"
-        /> */}
```
Comment (shivanigangadharan) [inline src/components/Header/Header.jsx:23]: can remove comments if not required

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const AuthContext = createContext({});
+
+const AuthProvider = ({ children }) => {
+
+    const getToken = localStorage.getItem("notesToken");
+
+    const authInitialState = {
+         loading:false,
+         error:null,
+         token: getToken ?? '',
```
Comment (shivanigangadharan) [inline src/context/auth-context.js:13]: I like this use of nullish coalescing operator.

File: src/components/Header/Header.jsx [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
+        <div>
+          <h3>NOTELY</h3>
         </div>
+      </div>
 
-            <Search/>
+      <Search />
       <div className="right_area">
-        <a href="#" className="login_btn" onClick={()=>setactive(prev=>!prev)}>Login</a>
+      {/* <a href="#" className="login_btn" onClick={() => setactive(prev => !prev)}>Login</a> */}
```
Comment (shivanigangadharan) [inline src/components/Header/Header.jsx]: can remove comment if not reqd.

File: src/Pages/NotesPage/NotesPage.jsx [hunk 4]
```diff
@@ -1,13 +1,16 @@
 import "./NotesPage.css";
 import {Header,Sidebar,Footer} from "../../components/index";
 import { MainArea } from "./MainArea/MainArea";
+import { ToastContainer} from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
 export const NotesPage=()=>{
     return(
         <>
         <input type="checkbox" id="check"></input>
```
Comment (shivanigangadharan) [inline src/Pages/NotesPage/NotesPage.jsx]: as input is not a container tag you can simply write `<input type="checkbox" id="check"/>` (without the closing tag)

**Review by shivanigangadharan — APPROVED** [review]
Code looks good, do add a deploy preview later on. Can merge this 👍

**Review by shivanigangadharan — APPROVED** [review]

**Review by tewarimohit — APPROVED** [review]
Looks good to merge...

File: src/components/loginPanel/helper.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  };
+  export const testLogin = {
+    email: "test@gmail.com",
+    password: "Test123@",
+  };
+  
+  export const validateInput=(values)=>{
+    const errors = {};
+
+    const passRegex =
```
Comment (ArjunGTX) [inline src/components/loginPanel/helper.js]: convention is to use `UPPER_SNAKE_CASE` for RegExp. And all the regular expressions can be moved to a separate file

File: src/components/loginPanel/loginPanel.css [hunk 6]
```diff
@@ -5,9 +5,10 @@
     box-shadow: var(--box-shadow);
     padding:2rem;
     background: var(--white);
-    text-align: center;
+    /* text-align: center; */
```
Comment (ArjunGTX) [inline src/components/loginPanel/loginPanel.css]: Can be removed

**Pull request merged**